lib/mongodb.js

```diff
@@ -802,6 +820,47 @@ MongoDB.prototype.count = function count(model, where, options, callback) {
 };
 
 /**
+ * Replace properties for the model instance data
+ * @param {String} model The name of the model
+ * @param {*} id The instance id
+ * @param {Object} data The model data
+ * @param {Object} options The options object
+ * @param {Function} [cb] The callback function
+ */
+MongoDB.prototype.replace = function replace(model, id, data, options, cb) {
+  if (this.debug) debug('replace', model, id, data);
+  var oid = this.coerceId(model, id);
+  this.replaceWithOptions(model, oid, data, {upsert: false}, function(err, data) {
+    cb(err, data);
+  });
+};
+
+/**
+ * Update a model instance with id
+ * @param {String} model The name of the model
+ * @param {Object} id The id of the model instance
+ * @param {Object} data The property/value pairs to be updated or inserted if {upsert: true} is passed as options
+ * @param {Object} options The options you want to pass for update, e.g, {upsert: true}
+ * @callback {Function} [cb] Callback function
+ */
+MongoDB.prototype.replaceWithOptions = function(model, id, data, options, cb) {
+  this.execute(model, 'update', {_id: id}, data, options, function(err, info) {
+    if (this.debug) debug('updateWithOptions.callback', model, {_id: id}, data, err, info);
+    if (err)  return cb && cb(err);
+    var result;
+    var cbInfo = {};
+    if (info.result && info.result.n == 1) {
+      result = data;
+      result.id = id;
+      cbInfo['isNewInstance'] = (info.result.nModified === 0) ? true : false;
+    } else {
+      result = undefined;
+    }
+    cb && cb(err, result, cbInfo);
```

Comment:
@raymondfeng: I had forgot to add this `cbInfo` which is used for `replaceOrCreate` to see if it is a `isNewInstance` or not in juggler for `after save` operation hook...
CC:  @bajtos